src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import {
+  useState, ChangeEvent,
+} from 'react';
+
+type TodoFilterProps = {
+  onFilterChange: (status: string) => void;
+  onTextChange: (text: string) => void;
+};
+
+export const TodoFilter: React.FC<TodoFilterProps> = (
+  { onFilterChange, onTextChange },
+) => {
+  const [selectedStatus, setSelectedStatus] = useState('all');
```

Comment:
why you need the same state here? please remove it as you already have it in `App` component